### PR TITLE
add clients & services count

### PR DESCRIPTION
**Pull request by leeminju531**
This PR is for clients, service counts using it's name.
Related to https://github.com/ros2/ros2cli/pull/771

Signed-off-by: leeminju531 <dlalswn531@naver.com>

### CODE REVIEW

**Comment by fujitatomoya** [comment]
@leeminju531 if this is ready to review, please let me know, i am happy to do review.

**Comment by leeminju531** [comment]
@fujitatomoya , this is ready to review :blush:

**Review by fujitatomoya — APPROVED** [review]

File: rclcpp/include/rclcpp/rclcpp.hpp [hunk 1]
```diff
@@ -92,9 +92,11 @@
  *   - rclcpp::Event
  * - List topic names and types:
  *   - rclcpp::Node::get_topic_names_and_types()
- * - Get the number of publishers or subscribers on a topic:
+ * - Get the number of publishers, subscribers, clients, or services on a topic:
```
Comment (alsora) [inline rclcpp/include/rclcpp/rclcpp.hpp:95]:
```suggestion
 * - Get the number of publishers, subscribers, clients, or services on a topic or service:
```

this is a nitpick, but the comment seems outdated now

Comment (leeminju531) [reply rclcpp/include/rclcpp/rclcpp.hpp]: :smile: :laughing:

File: rclcpp/include/rclcpp/rclcpp.hpp [hunk 2]
```diff
@@ -92,9 +92,11 @@
  *   - rclcpp::Event
  * - List topic names and types:
  *   - rclcpp::Node::get_topic_names_and_types()
- * - Get the number of publishers or subscribers on a topic:
+ * - Get the number of publishers, subscribers, clients, or services on a topic or service:
  *   - rclcpp::Node::count_publishers()
  *   - rclcpp::Node::count_subscribers()
+ *   - rclcpp::Node::count_clients()
+ *   - rclcpp::Node::count_services()
```
Comment (clalancette) [inline rclcpp/include/rclcpp/rclcpp.hpp:99]: I'm going to suggest that we write this slightly differently:

```suggestion
 * - Get the number of publishers or subscribers on a topic:
 *   - rclcpp::Node::count_publishers()
 *   - rclcpp::Node::count_subscribers()
 * - Get the number of clients or servers on a service:
 *   - rclcpp::Node::count_clients()
 *   - rclcpp::Node::count_services()
```

**Review by clalancette — CHANGES_REQUESTED** [review]
I've left one minor thing to fix inline.

This likely also needs to be rebased; once those two things are fixed, I'm happy with this and I'll run CI on it.

File: rclcpp/include/rclcpp/rclcpp.hpp (hunk 2 above)
Comment (leeminju531) [reply rclcpp/include/rclcpp/rclcpp.hpp]: Fixed in https://github.com/ros2/rclcpp/pull/2072/commits/2e5a93207cfbba53a8ccc425cdd0fb371dd5f207.

**Comment by leeminju531** [comment]
Cool. those are fixed and rebased:smiley:.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19623/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14131/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20345/)

**Pull request merged**